ospfd/ospf_vty.c

```diff
@@ -8083,6 +8084,7 @@ static int ospf_vty_dead_interval_set(struct vty *vty, const char *interval_str,
 		return CMD_WARNING_CONFIG_FAILED;
 	}
 
+	reset_wait_time = seconds >= params->v_wait ? seconds - params->v_wait : seconds;
```

Comment:
Need comment here explaining what you are doing here. Please use C rather than C++ comments. This seems rather arbitrary on how you set the reset time. A side-effect is that it does check for the wait timer changing. Rather than this, I'd reset the timer to seconds - the amount of time the timer has been running for each neighbor.

Reply:
So what I mean is something like this:

    time_remaining =  event_timer_remain_second(oi->t_wait) - seconds; 
    EVENT_OFF(oi->t_wait);
    if (time_remaining > 0) {
          OSPF_ISM_TIMER_ON(oi->t_wait, ospf_wait_timer,
							  time_remaining);
    }

Reply:
> So what I mean is something like this:
> 
> ```
> time_remaining =  event_timer_remain_second(oi->t_wait) - seconds; 
> EVENT_OFF(oi->t_wait);
> if (time_remaining > 0)
>           OSPF_ISM_TIMER_ON(oi->t_wait, ospf_wait_timer,
> 	        		                time_remaining);
> else 
>           ospf_wait_timer(oi->t_wait); 
> ```

Should time_remaining be `time_remaining =  seconds - event_timer_remain_second(oi->t_wait);`?